subprojects/packagefiles/whisper_cpp/meson.build

```diff
@@ -0,0 +1,66 @@
+project('whisper.cpp', ['cpp', 'c'], version: '1.5.2')
+
+cc = meson.get_compiler('c')
+
+math = cc.find_library('m', required: false)
+threads = dependency('threads')
+
+if build_machine.system() == 'linux'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+
+    if not get_option('no_avx')
+        add_project_arguments('-mavx', language: 'c')
+    endif
+
+    if not get_option('no_avx2')
+        add_project_arguments('-mavx2', language: 'c')
+    endif
+
+    if not get_option('no_fma')
+        add_project_arguments('-mfma', language: 'c')
+    endif
+
+    if not get_option('no_f16C')
+        add_project_arguments('-mf16c', language: 'c')
+    endif
+elif build_machine.system() == 'windows'
+    add_project_arguments('/utf-8', language: 'c')
+
+    if not get_option('no_avx')
+        add_project_arguments('/arch:AVX', language: 'c')
+    endif
+
+    if not get_option('no_avx2')
+        add_project_arguments('/arch:AVX2', language: 'c')
+    endif
+
+endif
+
+ggml_lib = library(
+    'ggml',
+    'ggml.c',
+    'ggml-alloc.c',
+    'ggml-backend.c',
+    'ggml-quants.c',
+    dependencies: [math, threads],
+)
+ggml_dep = declare_dependency(
+    dependencies: [math, threads],
+    include_directories: '.',
+    link_with: ggml_lib,
+)
+
+whisper_lib = library(
+    'whisper',
+    'whisper.cpp',
+    dependencies: [ggml_dep],
+)
+whisper_cpp_dep = declare_dependency(
+    include_directories: '.',
+    dependencies: [ggml_dep],
+    link_with: whisper_lib,
```

Comment:
consider `install`ing it, adding `version` and `soversion`

Installing headers and pkg-config `*.pc` file would be a nice, as a stretch goal.